src/dm/DispatchManagerActions.cc

```diff
@@ -2249,7 +2442,750 @@ int DispatchManager::live_updateconf(std::unique_ptr<VirtualMachine> vm,
     vmm->trigger_update_conf(vm->get_oid());
 
     vmpool->update(vm.get());
-    vmpool->update_search(vm.get());
+
+    return 0;
+}
+
+/* -------------------------------------------------------------------------- */
+/* -------------------------------------------------------------------------- */
+
+int DispatchManager::attach_sg(int vid, int nicid, int sgid,
+                               const RequestAttributes& ra, string& error_str)
+{
+    ostringstream oss;
+
+    bool is_update = false;
+
+    // -------------------------------------------------------------------------
+    // Check action consistency:
+    // - Object exists VM, NIC and SG
+    // - State is compatible with operation
+    // -------------------------------------------------------------------------
+    if ( sgpool->exist(sgid) < 0 )
+    {
+        oss << "Could not attach SG " << sgid << " to VM, SG does not exist";
+        error_str = oss.str();
+
+        return -1;
+    }
+
+    if ( auto vm = vmpool->get(vid) )
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

We observed that you are using unintended operations or assignments that could lead to logical errors, vulnerabilities, or undefined behavior. To prevent these issues, ensure that you follow best practices, such as using explicit comparisons and clear operator usage, and using parentheses to clarify intent. Learn more - https://cvefeed.io/cwe/detail/cwe-480-use-of-incorrect-operator